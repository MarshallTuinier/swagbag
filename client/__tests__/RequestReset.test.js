import { mount } from "enzyme";
import wait from "waait";
import { MockedProvider } from "react-apollo/test-utils";
import toJSON from "enzyme-to-json";
import RequestResetForm from "../components/RequestResetForm";
import { REQUEST_RESET_MUTATION } from "../GraphQL";

const mocks = [
  {
    request: {
      query: REQUEST_RESET_MUTATION,
      variables: { email: "mtuinier@gmail.com" }
    },
    result: {
      data: { requestReset: { message: "success", __typename: "Message" } }
    }
  }
];

describe("<RequestResetForm />", () => {
  it("renders and matches snapshot", async () => {
    const wrapper = mount(
      <MockedProvider>
        <RequestResetForm />
      </MockedProvider>
    );
    const form = wrapper.find("form[data-test='form']");
    expect(toJSON(form)).toMatchSnapshot();
  });

  it("calls the mutation successfully", async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <RequestResetForm />
      </MockedProvider>
    );
    // Simulate typing
    wrapper.find("input").simulate("change", {
      target: { name: "email", value: "mtuinier@gmail.com" }
    });
    // Submit the form
    wrapper.find("form").simulate("submit");
    await wait();
    wrapper.update();
    expect(wrapper.find("p").text()).toContain(
      "Success! Check your email for a reset link!"
    );
  });
});
