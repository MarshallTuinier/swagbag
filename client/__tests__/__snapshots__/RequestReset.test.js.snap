// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestResetForm /> renders and matches snapshot 1`] = `
<form
  className="RequestResetForm__Form-sc-12kq0ag-0 eqXHoV"
  data-test="form"
  method="post"
  onSubmit={[Function]}
>
  <fieldset
    aria-busy={false}
    disabled={false}
  >
    <h2>
      Reset your Password
    </h2>
    <DisplayError
      error={Object {}}
    />
    <label
      htmlFor="email"
    >
      Email
    </label>
    <input
      name="email"
      onChange={[Function]}
      placeholder="Email"
      required={true}
      type="email"
      value=""
    />
    <button
      type="submit"
    >
      Request Reset!
    </button>
  </fieldset>
</form>
`;
